cli/schema_describe.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2023 Democratized Data Foundation
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package cli
+
+import (
+	"github.com/spf13/cobra"
+
+	"github.com/sourcenetwork/defradb/client"
+)
+
+func MakeSchemaDescribeCommand() *cobra.Command {
+	var name string
+	var root string
+	var versionID string
+
+	var cmd = &cobra.Command{
+		Use:   "describe",
+		Short: "View schema description.",
+		Long: `Introspect schema types.
+
+Example: view all schema
+  defradb client schema describe
+		
+Example: view schema by name
+  defradb client schema describe --name User
+		
+Example: view schema by root
+  defradb client schema describe --root bae123
+		
+Example: view schema by version id
+  defradb client schema describe --version bae123
+		`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			store := mustGetStoreContext(cmd)
+
+			var schemas []client.SchemaDescription
+			switch {
+			case versionID != "":
+				schema, err := store.GetSchemaByVersionID(cmd.Context(), versionID)
+				if err != nil {
+					return err
+				}
+				schemas = []client.SchemaDescription{schema}
+
+			case root != "":
+				s, err := store.GetSchemaByRoot(cmd.Context(), root)
+				if err != nil {
+					return err
+				}
+				schemas = s
+
+			case name != "":
```

Comment:
suggestion: It would be good to add a check after the switch to check if name and versionID were given that they are actually related like you did with collections.

Reply:
Yeah I was in two minds if this should behave like the collection stuff, this *is* different though, as they are not going to be acting on the collection (saving docs and the like). I'll revisit today and see what happens.

- [ ] Revisit multi param schema describe stuff

Reply:
Even if not acting on it, it would be weird to return a set of results event if the provided params make no sense.

Reply:
I'm going to close this thread as it should match whatever we decide in https://github.com/sourcenetwork/defradb/pull/2006#discussion_r1374836866